src/patches/parts/hotkeys.cc

```diff
@@ -257,90 +228,76 @@ void ScreenManager_Update_Hook(auto original, ScreenManager* _this)
+      for (auto i = 0; i < all_pre_scan_widgets->max_length; ++i) {
+        auto pre_scan_widget = il2cpp_get_array_element<PreScanTargetWidget>(all_pre_scan_widgets, i);
+        if (pre_scan_widget
+            && (pre_scan_widget->_visibilityController->_state == VisibilityState::Visible
+                || pre_scan_widget->_visibilityController->_state == VisibilityState::Show)) {
+          auto rewardsWidget = pre_scan_widget->_rewardsButtonWidget;
+          if (rewardsWidget->_rewardsController->_state != VisibilityState::Visible
+              && rewardsWidget->_rewardsController->_state != VisibilityState::Show) {
+            show_info_pending = 5;
+          } else {
+            rewardsWidget->_rewardsController->Hide();
+          }
         }
       }
     }
-    show_info_pending -= 1;
-  }
 
-  // Lets try to remove the pre-scan because we hit escape and it's visible
-  if (GetKeyDownInt(KeyCode::Escape)) {
-    if (DidHideViewers()) {
-      return;
+    // Did we not find a rewards widget in the previous frame?
+    if (show_info_pending > 0) {
+      auto all_pre_scan_widgets = ObjectFinder<PreScanTargetWidget>::GetAll();
+      for (auto i = 0; i < all_pre_scan_widgets->max_length; ++i) {
+        auto       pre_scan_widget  = il2cpp_get_array_element<PreScanTargetWidget>(all_pre_scan_widgets, i);
+        const auto pre_scan_visible = pre_scan_widget
+                                      && (pre_scan_widget->_visibilityController->_state == VisibilityState::Visible
+                                          || pre_scan_widget->_visibilityController->_state == VisibilityState::Show);
+        if (pre_scan_visible) {
+          auto       rewardsWidget          = pre_scan_widget->_rewardsButtonWidget;
+          const auto rewards_widget_visible = rewardsWidget->_rewardsController->_state == VisibilityState::Visible
+                                              || rewardsWidget->_rewardsController->_state == VisibilityState::Show;
+          if (!rewards_widget_visible) {
+            rewardsWidget->_rewardsController->Show(true);
+          }
+        }
+      }
+      show_info_pending -= 1;
     }
   }
-
-  // {
-  //  auto all_pre_scan_widgets = ObjectFinder<PreScanTargetWidget>::GetAll();
-  //  for (auto i = 0; i < all_pre_scan_widgets->max_length; ++i) {
-  //    auto pre_scan_widget = il2cpp_get_array_element<PreScanTargetWidget>(all_pre_scan_widgets, i);
-  //    if (pre_scan_widget
-  //        && (pre_scan_widget->_visibilityController->_state == VisibilityState::Visible
-  //            || pre_scan_widget->_visibilityController->_state == VisibilityState::Show)) {
-  //      pre_scan_widget->_rewardsButtonWidget->RewardsClicked();
-  //    }
-  //  }
-  //}
-
-  // Config::Get().Load();
-
-  original(_this);
 }
 
-template <typename T> inline bool DidHideViewersOfType()
+template <typename T> inline bool CanHideViewersOfType(bool shouldHide = false)
```

Comment:
Not a huge fan of changing function behavior entirely based on a boolean param. Makes it not super clear on the callsite what is actually happening in most situations.
Especially when a 'Can' becomes a 'Do'

Reply:
I split them out into Can/DidHideViewer functions.  I guess we aren't likely to change the logic inside, it just duplicates that part of the code... unless we template it further and pass in an anonymous function but that seems overkill.